clustermesh-apiserver/vmmanager.go

```diff
@@ -124,9 +128,9 @@ func nodeOverrideFromCEW(n *nodeTypes.RegisterNode, cew *ciliumv2.CiliumExternal
 	}
 
 	// Override cluster
-	nk.Cluster = cfg.clusterName
-	nk.ClusterID = cfg.clusterID
-	nk.Labels[k8sConst.PolicyLabelCluster] = cfg.clusterName
+	nk.Cluster = m.ClusterName
```

Comment:
Not sure if applicable here but if RegisterNode used CluserIDName vs separates might clean up our structs a bit more?

Reply:
I wouldn't personally push this refactoring that far, because of the number of consequential changes. The idea of this PR is mainly to uniform the flags registration, without touching the rest of the usages as much as possible.